Unable to build properly (in CI too) since Leakcanary dependence has been added to project

#[Bug]While building kiwix-android app in android studio found error:package com.squareup.leakcanary does not exist

### Environment
  * version  : android studio
  * device / operating system : ubuntu 16.04
  
### The Bug:While running the app of kiwix-android on emulator there is an error occurred saying package com.squareup.leakcanary does not exist.Then I tried some of the solutions given on stackoverflow and github and have done some changes in build.gradle but still issue is not resolved.
I have added debugcompile and releasecompile in dependecies of build.gradle,but after running still I got an error.It looks like this:
![screenshot from 2018-03-24 03-28-57](https://user-images.githubusercontent.com/16611326/37867240-2f7c79ac-2f8d-11e8-93a5-491e232d424b.png)


#### Steps to reproduce:

  1. Build kiwix-android app in android studio from github.
  2. Run kiwix app on emulator.

  I want to run this app on android studio to have a look on its working.

```
BUILD SUCCESSFUL in 3m 49s
83 actionable tasks: 83 executed
Exception in thread "Thread-15" java.lang.NoClassDefFoundError: com/android/utils/PathUtils$1
	at com.android.utils.PathUtils.deleteIfExists(PathUtils.java:47)
	at com.android.utils.PathUtils.lambda$addRemovePathHook$0(PathUtils.java:125)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.ClassNotFoundException: com.android.utils.PathUtils$1
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 3 more
Running tests
Waiting for results
Waiting for results
Waiting for results
Waiting for results
Waiting for results
Waiting for results
Waiting for results
Waiting for results
Waiting for results
Waiting for results
Waiting for results
Waiting for results
Waiting for results
Waiting for results
Waiting for results
Waiting for results
Waiting for results
Waiting for results
Waiting for results
Waiting for results
Traceback (most recent call last):
  File "./testdroid.py", line 28, in <module>
    raise AssertionError("Success ratio only: {}".format(ratio))
AssertionError: Success ratio only: 0.909091
```

@kelson42 Are you sure this has been happening after the dependencies were introduced ? 
Isn't this related to #518 . Looks very similar.

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -120,9 +120,19 @@ dependencies {
   androidTestImplementation "org.mockito:mockito-android:2.7.22"
 
   // Leak canary
-  debugImplementation 'com.squareup.leakcanary:leakcanary-android:1.5.4'
-  releaseImplementation 'com.squareup.leakcanary:leakcanary-android-no-op:1.5.4'
-  testImplementation 'com.squareup.leakcanary:leakcanary-android-no-op:1.5.4'
+  implementation 'com.squareup.leakcanary:leakcanary-android-no-op:1.5.4'
+  // Only enable leak canary in debug builds
+  configurations.all { config ->
+    if (config.name.contains('debug') || config.name.contains("Debug")) {
+      config.resolutionStrategy.eachDependency { details ->
+        if (details.requested.group == 'com.squareup.leakcanary' &&
+            details.requested.name == 'leakcanary-android-no-op') {
+          details.useTarget(group: details.requested.group, name: 'leakcanary-android',
+              version: details.requested.version)
+        }
+      }
+    }
+  }
 }
 
 // Set custom app import directory

